core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -101,6 +101,7 @@ import org.apache.spark.util.{Clock, SystemClock, ThreadUtils, Utils}
  */
 private[spark] class ExecutorAllocationManager(
     client: ExecutorAllocationClient,
+    var scheduler: TaskScheduler,
```

Comment:
In dynamic resource allocation design in spark, we have loose coupling between allocation manager and scheduler.
Please drop this reference - we have to work around this issue without relying on scheduler internals.

Instead, we can track the hosts where the speculative task should not run (which is the host where the current task for partition is running)